src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Private/JavaScriptMarshalImpl.Bind.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+namespace System.Runtime.InteropServices.JavaScript.Private
+{
+    internal unsafe partial class JavaScriptMarshalImpl
+    {
+#nullable disable
+        public static Func<T1, TResult> BindJSFunctionByName<T1, TResult>(string functionName)
+        {
+            var signature = stackalloc IntPtr[2] {
+                typeof(TResult).TypeHandle.Value,
+                typeof(T1).TypeHandle.Value,
+            };
+            var exceptionMessage = BindJSFunction(string.Intern(functionName), 1, (IntPtr)(&signature), out int jsFunctionHandle, out int isException);
```

Comment:
I don't like the pattern of marshaling exceptions as string return values, i think we should preserve the JS stack. Otherwise we end up losing a ton of useful info when an error crosses language boundaries

Reply:
What do you propose instead?

Reply:
Solver with message as MonoString* + JSHandle to the exception for the lazy stack trace